x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -392,23 +388,17 @@ static class MultiBucketsAggExtractor implements AggValueExtractor {
         public Object value(Aggregation agg, Map<String, String> fieldTypeMap, String lookupFieldPrefix) {
             MultiBucketsAggregation aggregation = (MultiBucketsAggregation) agg;
 
-            HashMap<String, Object> nested = new HashMap<>();
+            var subAggLookupFieldPrefix = lookupFieldPrefix.isEmpty() ? agg.getName() : lookupFieldPrefix + "." + agg.getName();
+            Map<String, Object> nested = Maps.newLinkedHashMapWithExpectedSize(aggregation.getBuckets().size());
```

Comment:
Could the `Maps.newLinkedHashMapWithExpectedSize` method be used in line 366 too?

Reply:
We spoke offline, capturing the convo here: the cases where we're using `new LinkedHashMap` is because we do not have the size available.

> do you think it's okay to add size() and isEmpty() to [InternalAggregations](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/search/aggregations/InternalAggregations.java#L60)?

> Sure, it makes perfect sense as InternalAggregations is logically a collection.
If you decide to introduce it, then please make it in a separate PR and send it for review to search/aggregations folks.
Or even better, ask them first if such a change is welcome.